andrey_n/homework_4/olx_scrap/olx/spiders/example.py

```diff
@@ -0,0 +1,86 @@
+import scrapy
+from scrapy.exceptions import CloseSpider
+
+from olx.items import OlxItem
+
+
+class OlxSpider(scrapy.Spider):
+    '''
+    olx spider for scraping data
+    '''
+
+    # SETTINGS
+    name = 'olx'
+    start_urls = [
+            'https://www.olx.ua/elektronika/'
+        ]
+    TARGET = 10
+    count = 0
+    # END
+
+    # Catalog parser
+    def parse(self, response):
+        '''
+        Pagination parser for page with items
+        '''
+
+        items = response.css('table#offers_table a.detailsLink::attr(href)').getall()
+        pagination = response.css("span.fleft a::attr(href)").getall()
+        urls_res = [url for url in set(items) if
+                    url.find('https://www.olx.ua/obyavlenie/') != -1]
+
+        while len(pagination):
+            while urls_res:
+                url = urls_res.pop()
+                yield scrapy.Request(url=url, callback=self.parse_item)
+
+            url_p = pagination.pop()
+            yield scrapy.Request(url=url_p, callback=self.parse)
+
+    # Item Parser
+    def parse_item(self, response):
+
+        if OlxSpider.TARGET == OlxSpider.count:  # throw error when target ready
+            raise CloseSpider('Exceeded maximum items')
+
+        # Data cleaner
+        cost = response.css('strong.pricelabel__value::text').get()
+        price = cost[:cost.rfind(' ')]
+        currency = cost[cost.rfind(' ') + 1:]
```

Comment:
what will happen in case of price in USD?

Reply:
I think so.

Reply:
???

Reply:
It must work